### ISSUE TITLE
Integrate with Cireson SCSM Portal Watchlist feature

### ISSUE THREAD

**Issue by AdhocAdam** [AdhocAdam/smletsexchangeconnector#34]
When users are found on the TO or CC lines, create new entries in the Cireson SCSM Portal for the Watchlist so those related users can see said Work Item.

**Comment by AdhocAdam** [comment]
Hey @jeff-lang, I know you brought this one up awhile ago but I'm currently experimenting with it and looking for some input. Just thinking outloud here, my initials thoughts would be to leverage some new keywords like [watch] and [stopwatch] in order to achieve this.

Then as far as the actual use case go you could email several with a [watch] in the email and then those individuals should they so choose could remove themselves (i.e. the sender) with a [stopwatch] keyword. Then again, it's possible you may want to blanket [stopwatch] several as opposed to making them individually send an email.

Thoughts?

**Comment by AdhocAdam** [comment]
Tentative plan is as follows - deploy two new keywords that follow the behavior of the other ones. In this regard, they are actions that must be invoked by a single user as opposed to blanket including users (plural). Potentially could be a subconfiguration/future enhancement.

## New Keywords (configurable)
- [watch]
- [stopwatch]

## New Functions (added alongside the other function declarations)
```powershell
function Add-CiresonWatchListUser ($userguid, $workitemguid)
{
    $addToWatchlistAPIurl = "api/V3/WorkItem/AddToWatchlist?workitemId=$workitemguid&userId=$userguid"
    if ($ciresonPortalWindowsAuth -eq $true)
    {
        $addToWatchlistResult = Invoke-RestMethod -Uri ($ciresonPortalServer+$addToWatchlistAPIurl) -UseDefaultCredentials -Method post
    }
    else
    {
        $addToWatchlistResult = Invoke-RestMethod -Uri ($ciresonPortalServer+$addToWatchlistAPIurl) -Method post -Headers @{"Authorization"=Get-CiresonPortalAPIToken}
    }
    return $addToWatchlistResult
}
function Remove-CiresonWatchListUser ($userguid, $workitemguid)
{
    $removeFromWatchlistAPIurl = "api/V3/WorkItem/DeleteFromWatchlist?workitemId=$workitemguid&userId=$userguid"
    if ($ciresonPortalWindowsAuth -eq $true)
    { 
        $removeFromWatchlistResult = Invoke-RestMethod -Uri ($ciresonPortalServer+$removeFromWatchlistAPIurl) -UseDefaultCredentials -Method DELETE
    }
    else
    {
        $removeFromWatchlistResult = Invoke-RestMethod -Uri ($ciresonPortalServer+$removeFromWatchlistAPIurl) -Method DELETE -Headers @{"Authorization"=Get-CiresonPortalAPIToken}
    }
    return $removeFromWatchlistResult
}
```

## Update-WorkItem changes
Two new actions would be introduced within relevant Work Item/user blocks. Need to add validation to ensure the portal option was configured.
```powershell
"\[$addWatchlistKeyword]" {$cpu = Get-CiresonPortalUser -username $commentLeftBy.UserName -domain $commentLeftBy.Domain; if ($cpu) {Add-CiresonWatchListUser -userguid $cpu.Id -workitemguid $workItem.__InternalId} }
"\[$removeWatchlistKeyword]" {$cpu = Get-CiresonPortalUser -username $commentLeftBy.UserName -domain $commentLeftBy.Domain; if ($cpu) {Remove-CiresonWatchListUser -userguid $cpu.Id -workitemguid $workItem.__InternalId} }
```

**Comment by AdhocAdam** [comment]
Feature will be made available on next releases through v1.7 and v2.1
